exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/javatime/JavaDateColumnType.kt

```diff
@@ -127,7 +127,14 @@ class JavaLocalDateColumnType : ColumnType<LocalDate>(), IDateColumnType {
 
     override fun nonNullValueToString(value: LocalDate): String {
         val instant = Instant.from(value.atStartOfDay(ZoneId.systemDefault()))
-        return "'${DEFAULT_DATE_STRING_FORMATTER.format(instant)}'"
+        val formatted = DEFAULT_DATE_STRING_FORMATTER.format(instant)
+        if (currentDialect is OracleDialect) {
+            // Date literal in Oracle DB must match NLS_DATE_FORMAT parameter.
+            // That parameter can be changed on DB level.
+            // But format can be also specified per literal with TO_DATE function
+            return "TO_DATE('$formatted', 'YYYY-MM-DD')"
+        }
+        return "'$formatted'"
```

Comment:
This will end up affecting all defaults (as you've found) and insert/update values (generated in SQL string at least), but I agree that it seems the only way to move forward this issue.